drivers/sensor/iis2dh/iis2dh.c

```diff
@@ -0,0 +1,315 @@
+	},
+	{
+		/* NM mode */
+		3910/64,	/* 2G */
+		7810/64,	/* 4G */
+		15630/64,	/* 8G */
+		46950/64,	/* 16G */
+	},
+	{
+		/* LP mode */
+		15630/256,	/* 2G */
+		31250/256,	/* 4G */
+		62500/256,	/* 8G */
+		188680/256,	/* 16G */
+	},
+};
+
+static int iis2dh_set_fs_raw(struct iis2dh_data *iis2dh, uint8_t fs)
+{
+	int err;
+
+	err = iis2dh_full_scale_set(iis2dh->ctx, fs);
+
+	if (!err) {
+		/* save internally gain for optimization */
+		iis2dh->gain = iis2dh_gain[IIS2DH_HR_12bit][fs];
+	}
+
+	return err;
+}
+
+#if (CONFIG_IIS2DH_RANGE == 0)
+/**
+ * iis2dh_set_range - set full scale range for acc
+ * @dev: Pointer to instance of struct device (I2C or SPI)
+ * @range: Full scale range (2, 4, 8 and 16 G)
+ */
+static int iis2dh_set_range(struct device *dev, uint16_t range)
+{
+	int err;
+	struct iis2dh_data *iis2dh = dev->driver_data;
+	uint8_t fs = IIS2DH_FS_TO_REG(range);
+
+	err = iis2dh_set_fs_raw(iis2dh, fs);
+
+	return err;
+}
+#endif
+
+#if (CONFIG_IIS2DH_ODR == 0)
+/**
+ * iis2dh_set_odr - set new sampling frequency
+ * @dev: Pointer to instance of struct device (I2C or SPI)
+ * @odr: Output data rate
+ */
+static int iis2dh_set_odr(struct device *dev, uint16_t odr)
+{
+	struct iis2dh_data *iis2dh = dev->driver_data;
+	const struct iis2dh_device_config *cfg = dev->config_info;
+	iis2dh_odr_t val;
+
+	val = IIS2DH_ODR_TO_REG_HR(cfg->pm, odr);
+
+	return iis2dh_data_rate_set(iis2dh->ctx, val);
+}
+#endif
+
+static inline void iis2dh_convert(struct sensor_value *val, int raw_val,
+				  uint32_t gain)
+{
+	int64_t dval;
+
+	/* Gain is in ug/LSB */
+	/* Convert to m/s^2 */
+	dval = ((int64_t)raw_val * gain * SENSOR_G) / 1000000LL;
+	val->val1 = dval / 1000000LL;
+	val->val2 = dval % 1000000LL;
+}
+
+static inline void iis2dh_channel_get_acc(struct device *dev,
+					  enum sensor_channel chan,
+					  struct sensor_value *val)
+{
+	int i;
+	uint8_t ofs_start, ofs_stop;
+	struct iis2dh_data *iis2dh = dev->driver_data;
+	struct sensor_value *pval = val;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		ofs_start = ofs_stop = 0U;
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		ofs_start = ofs_stop = 1U;
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		ofs_start = ofs_stop = 2U;
+		break;
+	default:
+		ofs_start = 0U; ofs_stop = 2U;
+		break;
+	}
+
+	for (i = ofs_start; i <= ofs_stop ; i++) {
+		iis2dh_convert(pval++, iis2dh->acc[i], iis2dh->gain);
+	}
+}
+
+static int iis2dh_channel_get(struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		iis2dh_channel_get_acc(dev, chan, val);
+		return 0;
+	default:
+		LOG_DBG("Channel not supported");
+		break;
+	}
+
+	return -ENOTSUP;
+}
+
+static int iis2dh_config(struct device *dev, enum sensor_channel chan,
+			 enum sensor_attribute attr,
+			 const struct sensor_value *val)
+{
+	switch (attr) {
+#if (CONFIG_IIS2DH_RANGE == 0)
```

Comment:
Does Zephyr attempt to reduce use of #ifdef?

It sems like this could use if() instead